llvm/lib/Target/DirectX/DXILDataScalarization.cpp

```diff
@@ -117,19 +117,23 @@ DataScalarizerVisitor::lookupReplacementGlobal(Value *CurrOperand) {
   return nullptr; // Not found
 }
 
-static bool isArrayOfVectors(Type *T) {
+// Helper function to check if a type is a vector or an array of vectors
+static bool isVectorOrArrayOfVectors(Type *T) {
+  if (isa<VectorType>(T))
+    return true;
```

Comment:
might need to remove these two line.

Reply:
After talking with the team the `shuffle.ll` test is not valid DXIL. The test is just to make sure the bitcodewriter can handle shuffle vector instructions potentially for future use. This means no code change is necessary.